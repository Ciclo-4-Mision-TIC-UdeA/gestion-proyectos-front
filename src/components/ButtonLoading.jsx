import React from 'react';
import ReactLoading from 'react-loading';

const ButtonLoading = ({ disabled, loading, text }) => {
  return (
    <button
      disabled={disabled}
      type='submit'
      className='bg-indigo-700 text-white font-bold text-lg py-3 px-6  rounded-xl hover:bg-indigo-500 shadow-md my-2 disabled:opacity-50 disabled:bg-gray-700'
    >
      {loading ? <ReactLoading type='spin' height={30} width={30} /> : text}
    </button>
  );
};

export default ButtonLoading;
