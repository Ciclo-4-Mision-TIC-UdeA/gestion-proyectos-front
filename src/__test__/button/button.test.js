import React from 'react';
import ButtonLoading from 'components/ButtonLoading';
import { render, cleanup, screen } from '@testing-library/react';

afterEach(cleanup);
it('renders okay', () => {
  render(<ButtonLoading loading={false} text={'Hola'} />);
  expect(screen.getByTestId('button-test')).toBeInTheDocument();
});

it('renders button correctly without loading', () => {
  render(<ButtonLoading loading={false} text={'Hola'} />);
  expect(screen.getByTestId('button-test')).toHaveTextContent('Hola');
});

it('renders button correctly with loading', () => {
  render(<ButtonLoading loading={true} text={'Hola'} />);
  expect(screen.getByTestId('button-test')).not.toHaveTextContent('Hola');
});

it('renders button correctly disabled', () => {
  render(<ButtonLoading disabled={true} loading={false} text={'Hola'} />);
  expect(screen.getByTestId('button-test')).toHaveAttribute('disabled');
});

it('renders button correctly enabled', () => {
  render(<ButtonLoading disabled={false} loading={false} text={'Hola'} />);
  expect(screen.getByTestId('button-test')).not.toHaveAttribute('disabled');
});

it('renders loading', () => {
  render(<ButtonLoading loading={true} disabled={false} text={'Hola'} />);
  expect(screen.getByTestId('button-loading-test')).toBeInTheDocument();
});

it('renders svg when loading is activated', () => {
  render(<ButtonLoading disabled={false} loading={true} text={'Hola'} />);
  expect(screen.getByTestId('button-test')).toMatchSnapshot();
});
