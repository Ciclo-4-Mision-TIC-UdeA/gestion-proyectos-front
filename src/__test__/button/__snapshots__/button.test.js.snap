// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders svg when loading is activated 1`] = `
<button
  class="bg-indigo-700 text-white font-bold text-lg py-3 px-6  rounded-xl hover:bg-indigo-500 shadow-md my-2 disabled:opacity-50 disabled:bg-gray-700"
  data-testid="button-test"
  type="submit"
>
  <div
    data-testid="button-loading-test"
    style="fill: #fff; height: 30px; width: 30px;"
  >
    <svg
      viewBox="0 0 32 32"
      xmlns="http://www.w3.org/2000/svg"
    >
      
  
      <path
        d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4"
        opacity=".25"
      />
      
  
      <path
        d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z"
      >
        
    
        <animatetransform
          attributeName="transform"
          dur="0.8s"
          from="0 16 16"
          repeatCount="indefinite"
          to="360 16 16"
          type="rotate"
        />
        
  
      </path>
      

    </svg>
    

  </div>
</button>
`;
